File: packages/java/src/main/java/com/peruspatial/territorial/GenerateRecommendations.java
package com.peruspatial.territorial;

import java.util.ArrayList;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.Set;

public final class GenerateRecommendations {
  private GenerateRecommendations() {}

  public static List<String> generateRecommendations(List<String> entities) {
    List<String> out = new ArrayList<>();
    if (entities == null) return out;
    Set<String> seen = new LinkedHashSet<>();
    for (String entity : entities) {
      if (seen.add(entity)) {
        out.add(RecommendationFor.recommendationFor(entity));
      }
    }
    return out;
  }
}

File: packages/go/generate_recommendations.go
package perspatial

func GenerateRecommendations(entities []string) []string {
	seen := make(map[string]struct{}, len(entities))
	out := make([]string, 0, len(entities))
	for _, e := range entities {
		if _, ok := seen[e]; ok {
			continue
		}
		seen[e] = struct{}{}
		out = append(out, RecommendationFor(e))
	}
	return out
}
